crates/iceberg-catalog/src/api/management/v1/warehouse/undrop.rs

```diff
@@ -0,0 +1,120 @@
+use crate::api::iceberg::types::PageToken;
+use crate::api::iceberg::v1::PaginationQuery;
+use crate::api::management::v1::warehouse::{UndropTabularsRequest, UndropTarget};
+use crate::request_metadata::RequestMetadata;
+use crate::service::authz::Authorizer;
+use crate::service::{Catalog, ListFlags, TableIdentUuid, TabularIdentUuid, Transaction};
+use crate::{api, WarehouseIdent};
+use iceberg_ext::catalog::rest::ErrorModel;
+use itertools::Itertools;
+
+pub(crate) async fn collect_tabular_ids<C: Catalog>(
+    warehouse_ident: WarehouseIdent,
+    request: UndropTabularsRequest,
+    transaction: <C::Transaction as Transaction<C::State>>::Transaction<'_>,
+) -> api::Result<Vec<TableIdentUuid>> {
+    Ok(match request.target {
+        UndropTarget::Tabulars { tabulars: tabs } => {
+            tabs.into_iter().map(|i| TableIdentUuid::from(*i)).collect()
+        }
+        UndropTarget::Namespace { namespace: ns } =>
+        // TODO: do we want to paginate here?
```

Comment:
I think it would be the most correct way to paginate. Alternatively for now we could also just document that this undroppes the first 1000 tables so the namespace and the client should retry.

I tend towards implementing pagination now as we will need pagination at least for "drop cascade" later, and it would be strange if one API paginates and the other doesn't. On the other hand this is a sync API and drop might be asynchronous?

Reply:
removed namespace undrop target as discussed